deploy/charts/cert-manager/README.template.md

```diff
@@ -74,120 +78,23 @@ The following table lists the configurable parameters of the cert-manager chart
-| `webhook.readinessProbe.initialDelaySeconds` | The readiness probe initial delay (in seconds) | `5` |
-| `webhook.readinessProbe.periodSeconds` | The readiness probe period (in seconds) | `5` |
-| `webhook.readinessProbe.successThreshold` | The readiness probe success threshold | `1` |
-| `webhook.readinessProbe.timeoutSeconds` | The readiness probe timeout (in seconds) | `1` |
-| `cainjector.enabled` | Toggles whether the cainjector component should be installed (required for the webhook component to work) | `true` |
-| `cainjector.replicaCount` | Number of cert-manager cainjector replicas | `1` |
-| `cainjector.podAnnotations` | Annotations to add to the cainjector pods | `{}` |
-| `cainjector.podLabels` | Labels to add to the cert-manager cainjector pod | `{}` |
-| `cainjector.deploymentAnnotations` | Annotations to add to the cainjector deployment | `{}` |
-| `cainjector.extraArgs` | Optional flags for cert-manager cainjector component | `[]` |
-| `cainjector.serviceAccount.create` | If `true`, create a new service account for the cainjector component | `true` |
-| `cainjector.serviceAccount.name` | Service account for the cainjector component to be used. If not set and `cainjector.serviceAccount.create` is `true`, a name is generated using the fullname template |  |
-| `cainjector.serviceAccount.annotations` | Annotations to add to the service account for the cainjector component |  |
-| `cainjector.serviceAccount.automountServiceAccountToken` | Automount API credentials for the cainjector Service Account | `true` |
-| `cainjector.resources` | CPU/memory resource requests/limits for the cainjector pods | `{}` |
-| `cainjector.nodeSelector` | Node labels for cainjector pod assignment | `{}` |
-| `cainjector.affinity` | Node affinity for cainjector pod assignment | `{}` |
-| `cainjector.tolerations` | Node tolerations for cainjector pod assignment | `[]` |
-| `cainjector.image.repository` | cainjector image repository | `quay.io/jetstack/cert-manager-cainjector` |
-| `cainjector.image.tag` | cainjector image tag | `{{RELEASE_VERSION}}` |
-| `cainjector.image.pullPolicy` | cainjector image pull policy | `IfNotPresent` |
-| `cainjector.securityContext` | Security context for cainjector pod assignment | `{}` |
-| `cainjector.containerSecurityContext` | Security context to be set on cainjector component container | `{}` |
+{{- $valueComments := comments "Values" }}
+{{- range $index, $element := $valueComments }}
+{{- $parts := regexSplit "Example:\n" ($element.Comment | replace "|" "&#124;") -1 }}
+{{- $text := first $parts | replace "\n" " " | regexReplaceAll " *" " " }}
+{{- $examples := list }}
+{{- range $example := rest $parts }}
+{{- $examples = append $examples (list "<pre lang=\"yaml\">" ($example | replace "\n" "&NewLine;") "</pre>" | join "") }}
+{{- end}}
+| `{{ join "." $element.Path }}` | {{ join "" (prepend $examples $text) }} | `{{ $element.Value }}` |
+{{- end }}
```

Comment:
I honestly can't decipher this. Why can't `tem` have a function called `TableOfContentsMarkdown` which does all this?

Reply:
It can; I just think that it would cause a lot of upstream change requests. Now, all the logic is implemented in the template (identical to how a helm template would solve this). If a change in the comment > markdown logic is required, it can be done directly by updating the template file.

Reply:
In the latest version, I removed the "example" parsing logic; the json schema should more clearly indicate the structure + this caused the table to become even larger.
Maybe, this can be re-added, but that will not be part of this PR.